app/models/application_record.rb

```diff
@@ -1,5 +1,13 @@
 # frozen_string_literal: true
 
 class ApplicationRecord < ActiveRecord::Base
+  after_initialize :slugify_name
```

Comment:
Gostei da ideia de vc pensar em colocar isso aqui já que é algo usado em todos os models, e agora ele faz sentido porque agora todos os models tem um slug, mas se a gente for pensar que no futuro possivelmente iremos criar algum model que não tenha slug, aí esse código vai dar uma atrapalhada no rolê 😄 

Então o que eu faria é manter esse método `slugify_name` aqui, mas jogar esse `after_initialize` pra dentro de todos os models que precisam dele.

Reply:
Se bem que no método `slugify_name` vc depende que todos os models também tenham um atributo `name`. Ou seja, com esse código aqui, vc bota uma dependência de que todos os seus models tenham `slug` e `name`, né?

Então talvez pra evitar de duplicar muito código, o que eu faria seria, talvez, algo assim:

No model DefaultBill:
```rb
after_initialize :slugify_name

private

def slugify_name
  slugify(name)
end
```

E aqui no ApplicationRecord:
```rb
protected

def slugify(value)
  self.slug = "#{value}-#{SecureRandom.hex(3)}".downcase.parameterize if value.present?
end
```

Algo assim... não consegui pensar em nada melhor agora 😅